src/coreclr/jit/importer.cpp

```diff
@@ -14411,15 +14411,27 @@ void Compiler::impImportBlockCode(BasicBlock* block)
                     impStackTop(0);
                 }
 
-                // If the expression to dup is simple, just clone it.
-                // Otherwise spill it to a temp, and reload the temp
-                // twice.
                 StackEntry se   = impPopStack();
                 GenTree*   tree = se.val;
                 tiRetVal        = se.seTypeInfo;
                 op1             = tree;
 
-                if (!opts.compDbgCode && !op1->IsIntegralConst(0) && !op1->IsFPZero() && !op1->IsLocal())
+                // If the expression to dup is simple, just clone it.
+                // Otherwise spill it to a temp, and reload the temp twice.
+                bool needsTemp = true;
+                if (opts.OptimizationEnabled())
+                {
+                    if (op1->IsIntegralConst(0) || op1->IsFPZero() || op1->IsLocal())
+                    {
+                        needsTemp = false;
+                    }
+                    else if (tree->TypeIs(TYP_BYREF) && tree->OperIs(GT_ADDR) && tree->gtGetOp1()->IsLocal())
```

Comment:
Should this also check for `TYP_I_IMPL` or `GT_IND`? What about the `GT_OBJ` scenario?

Asking since these are all special cases we handle in `impSIMDPopStack` to cover `ref T`, `T*`, and a few other scenarios: https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/simd.cpp#L1225-L1305

So I'd imagine they may also qualify here in some circumstances.

Reply:
This is checking for addresses only, so, no need to check for indirections.

This should probably be using `impIsAddressInLocal` to catch both `LCL_VAR_ADDR` and `LCL_FLD_ADDR` - equivalent trees.